tox_conda/plugin.py

```diff
@@ -1,19 +1,187 @@
+import json
 import logging
+import os
+import shutil
+import sys
+from warnings import warn
 
 import pluggy
 
 hookimpl = pluggy.HookimplMarker("tox")
-log = logging.getLogger('conda')
+log = logging.getLogger("conda")
+
+
+on_win = bool(sys.platform == "win32")
 
 
 @hookimpl
 def tox_addoption(parser):
-    """Add a command line option for later use"""
-    parser.add_argument(
-        '--my-opt', action='store', help='my custom option')
+    parser.add_testenv_attribute(
+        name="conda_channels",
+        type="line-list",
+        default=("defaults"),
+        help="Conda channels for environment creation.",
+    )
+
+
+@hookimpl
+def tox_get_python_executable(envconfig):
+    """Return a python executable for the given python base name."""
+    if on_win:
+        return os.path.join(envconfig.envdir, "python.exe")
+    else:
+        return os.path.join(envconfig.envdir, "python")
 
 
 @hookimpl
 def tox_configure(config):
-    """Access your option during configuration"""
-    log.info("my option is: '%s'", config.option.my_opt)
+    """Called after command line options are parsed and ini-file has been read."""
+
+    # Directory to store conda files.
+    config.conda_workdir = os.path.join(config.toxworkdir, "conda")
+
+    # Override getting python executable.
+    envconfig_class = type(config.envconfigs[config.envlist[0]])
+    envconfig_class.get_envpython = get_conda_python
+
+    # Set testenv configurations.
+    envs = config.envlist[:]
+    for env in envs:
+        # Get envconfig for this env
+        envconfig = config.envconfigs[env]
+
+        # Only set conda to True for supported Python versions, others are created normally.
+        basepython = envconfig.basepython
+        if not basepython.startswith("python"):
+            msg = "{0} will not be created using conda, only Python basepython is supported by tox-conda.".format(
+                env
+            )
+            warn(RuntimeWarning(msg))
+            envconfig.conda = False
+            continue
+        envconfig.conda = True
+
+        # Split Conda dependencies from pip dependencies and create environment.yaml.
+        split_deps(envconfig)
+        create_env_yml(envconfig)
+
+
+@hookimpl
+def tox_testenv_create(venv, action):
+    """Perform creation action for this venv."""
+
+    if not venv.envconfig.conda:
+        return None
+
+    env_location = str(venv.path)
+    yaml = venv.envconfig.conda_envyaml
+    redirect = venv.session.config.option.verbose_level < 2
+    result = []
+
+    # Remove old environment if recreating.
+    if venv.envconfig.recreate:
+        if os.path.exists(env_location):
+            shutil.rmtree(env_location)
```

Comment:
I'm believe this logic is actually unnecessary, since `tox_testenv_create` appears to only be called if the environment doesn't already exist and if `-r/--recreate` wasn't given. In this case, it makes the most sense for the `conda` command below to be `create` instead of `update`.